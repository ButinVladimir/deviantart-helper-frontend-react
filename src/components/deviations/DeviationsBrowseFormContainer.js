import { connect } from 'react-redux';
import deviationsChangeTitleActionCreator from '../../redux/action-creators/deviations/change-title';
import deviationsChangePublishedTimeBeginActionCreator from '../../redux/action-creators/deviations/change-published-time-begin';
import deviationsChangePublishedTimeEndActionCreator from '../../redux/action-creators/deviations/change-published-time-end';
import deviationsChangeSortFieldActionCreator from '../../redux/action-creators/deviations/change-sort-field';
import deviationsChangeSortOrderActionCreator from '../../redux/action-creators/deviations/change-sort-order';
import deviationsBrowseActionCreator from '../../redux/action-creators/deviations/browse';
import deviationsBrowsePrevPageActionCreator from '../../redux/action-creators/deviations/browse-prev-page';
import deviationsBrowseNextPageActionCreator from '../../redux/action-creators/deviations/browse-next-page';
import DeviationsBrowseForm from './DeviationsBrowseForm';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  page: state.deviations.pageBrowse,
  sortField: state.deviations.sortField,
  sortOrder: state.deviations.sortOrder,
  title: state.deviations.title,
  publishedTimeBegin: state.deviations.publishedTimeBegin,
  publishedTimeEnd: state.deviations.publishedTimeEnd,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  titleChangeHandler: e => dispatch(deviationsChangeTitleActionCreator(e.target.value)),
  publishedTimeBeginChangeHandler:
    e => dispatch(deviationsChangePublishedTimeBeginActionCreator(e.target.value)),
  publishedTimeEndChangeHandler:
    e => dispatch(deviationsChangePublishedTimeEndActionCreator(e.target.value)),
  sortFieldChangeHandler: e => dispatch(deviationsChangeSortFieldActionCreator(e.target.value)),
  sortOrderChangeHandler: e => dispatch(deviationsChangeSortOrderActionCreator(e.target.value)),
  submitHandler: () => dispatch(deviationsBrowseActionCreator(ownProps.config)),
  prevPageHandler: () => dispatch(deviationsBrowsePrevPageActionCreator(ownProps.config)),
  nextPageHandler: () => dispatch(deviationsBrowseNextPageActionCreator(ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowseForm);
