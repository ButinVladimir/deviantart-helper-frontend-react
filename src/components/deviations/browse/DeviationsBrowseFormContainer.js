import { connect } from 'react-redux';
import deviationsBrowseChangeTitleActionCreator from '../../../redux/action-creators/deviations/browse/change-title';
import deviationsBrowseChangePublishedTimeBeginActionCreator from '../../../redux/action-creators/deviations/browse/change-published-time-begin';
import deviationsBrowseChangePublishedTimeEndActionCreator from '../../../redux/action-creators/deviations/browse/change-published-time-end';
import deviationsBrowseChangeSortFieldActionCreator from '../../../redux/action-creators/deviations/browse/change-sort-field';
import deviationsBrowseChangeSortOrderActionCreator from '../../../redux/action-creators/deviations/browse/change-sort-order';
import deviationsBrowseLoadPageActionCreator, { loadPage } from '../../../redux/action-creators/deviations/browse/load-page';
import DeviationsBrowseForm from './DeviationsBrowseForm';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  page: state.deviations.browse.page,
  pageCount: state.deviations.browse.pageCount,
  sortField: state.deviations.browse.sortField,
  sortOrder: state.deviations.browse.sortOrder,
  title: state.deviations.browse.title,
  publishedTimeBegin: state.deviations.browse.publishedTimeBegin,
  publishedTimeEnd: state.deviations.browse.publishedTimeEnd,
  pageLoading: state.deviations.browse.pageLoading,
  showPagination: state.deviations.browse.showPagination,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  titleChangeHandler: e => dispatch(deviationsBrowseChangeTitleActionCreator(e.target.value)),
  publishedTimeBeginChangeHandler:
    value => dispatch(deviationsBrowseChangePublishedTimeBeginActionCreator(value)),
  publishedTimeEndChangeHandler:
    value => dispatch(deviationsBrowseChangePublishedTimeEndActionCreator(value)),
  sortFieldChangeHandler:
    e => dispatch(deviationsBrowseChangeSortFieldActionCreator(e.target.value)),
  sortOrderChangeHandler:
    e => dispatch(deviationsBrowseChangeSortOrderActionCreator(e.target.value)),
  submitHandler: () => dispatch(deviationsBrowseLoadPageActionCreator(ownProps.config)),
  // Pagination in Bulma starts from 1 while pagination on backend start from 0.
  loadPageHandler: page => dispatch(loadPage(page - 1, ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowseForm);
