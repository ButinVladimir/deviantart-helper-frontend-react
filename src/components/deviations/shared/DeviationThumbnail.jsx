import React from 'react';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Tag from 'react-bulma-components/lib/components/tag';

export default function DeviationThumbnail({
  title,
  thumbnail,
  url,
  publishedTime,
  nsfw,
}) {
  const publishedTimeDate = new Date(publishedTime);

  return (
    <Media.Item position="left" className="image-container">
      <div>
        <img
          alt={title}
          src={thumbnail.src}
        />
      </div>
      <div>
        <small className="has-text-grey">
          {publishedTimeDate.toLocaleString()}
        </small>
      </div>
      <div>
        <a href={url}>Visit DA page</a>
      </div>
      {nsfw && <Tag color="danger">NSFW</Tag>}
    </Media.Item>
  );
}

DeviationThumbnail.propTypes = {
  title: PropTypes.string.isRequired,
  thumbnail: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  url: PropTypes.string.isRequired,
  publishedTime: PropTypes.number.isRequired,
  nsfw: PropTypes.bool.isRequired,
};
