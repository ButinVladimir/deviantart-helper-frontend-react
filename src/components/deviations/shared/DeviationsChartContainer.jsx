import { connect } from 'react-redux';
import Chart from '../../chart/Chart';
import * as deviationChart from '../../../redux/action-creators/deviations/chart/change-form-field-values';
import { dataSetOptions } from '../../../consts/data-sets';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The Redux state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  dataSetTitlesMap: new Map(dataSetOptions),
  dataSet: state.deviations.chart.dataSet,
  roundPeriod: state.deviations.chart.roundPeriod,
  showTime: state.deviations.chart.showTime,
  showDifferences: state.deviations.chart.showDifferences,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The Redux dispatch.
 * @returns {Object} Props.
 */
const mapDispatchToProps = dispatch => ({
  dataSetChangeHandler:
    e => dispatch(deviationChart.changeDataSetActionCreator(e.target.value)),
  roundPeriodChangeHandler:
    e => dispatch(deviationChart.changeRoundPeriodActionCreator(e.target.value)),
  showTimeChangeHandler:
    e => dispatch(deviationChart.changeShowTimeActionCreator(e.target.checked)),
  showDifferencesChangeHandler:
    e => dispatch(deviationChart.changeShowDifferencesActionCreator(e.target.checked)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Chart);
