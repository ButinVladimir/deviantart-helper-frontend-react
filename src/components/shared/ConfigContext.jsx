import React, { createContext } from 'react';

const ConfigContext = createContext(null);

/**
 * @description
 * Gets config context provider.
 *
 * @returns {import('react').Provider} Provider.
 */
export const getConfigProvider = () => ConfigContext.Provider;

/**
 * @description
 * HOC to inject config.
 *
 * @param {React} Component - React component.
 * @returns {(props: Object) => React} Functional component.
 */
export default Component => props => (
  <ConfigContext.Consumer>
    {config => <Component config={config} {...props} />}
  </ConfigContext.Consumer>
);
