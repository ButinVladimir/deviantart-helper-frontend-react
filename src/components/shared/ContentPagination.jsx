import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Pagination from 'react-bulma-components/lib/components/pagination';

export default function ContentPagination({
  page,
  pageCount,
  showPagination,
  loadPageHandler,
}) {
  return (
    <Section>
      <Container>
        {showPagination && (
          <Pagination
            // Pagination in Bulma starts from 1 while pagination on backend start from 0.
            current={page + 1}
            total={pageCount}
            delta={3}
            onChange={loadPageHandler}
          />
        )}
      </Container>
    </Section>
  );
}

ContentPagination.propTypes = {
  page: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  showPagination: PropTypes.bool.isRequired,
  loadPageHandler: PropTypes.func.isRequired,
};
