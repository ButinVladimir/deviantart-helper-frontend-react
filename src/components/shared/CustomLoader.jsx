import React from 'react';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Level from 'react-bulma-components/lib/components/level';
import Loader from 'react-bulma-components/lib/components/loader';

export default function CustomLoader() {
  return (
    <Section>
      <Container>
        <Level>
          <Level.Item>
            <Loader className="custom-loader" />
          </Level.Item>
        </Level>
      </Container>
    </Section>
  );
}
