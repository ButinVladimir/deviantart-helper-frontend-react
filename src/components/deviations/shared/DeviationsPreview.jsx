import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import { Field, Control, Checkbox } from 'react-bulma-components/lib/components/form';
import DeviationThumbnail from './DeviationThumbnail';
import StatisticsLevel from '../../shared/StatisticsLevel';
import { DEVIATIONS_DETAILS, ID_PARAM } from '../../../consts/routes';

export default function DeviationsPreview({
  thumbnail,
  title,
  url,
  id,
  publishedTime,
  nsfw,
  views,
  favourites,
  comments,
  downloads,
  selected,
  toggleSelectionHandler,
}) {
  return (
    <Media>
      <DeviationThumbnail
        title={title}
        thumbnail={thumbnail}
        url={url}
        publishedTime={publishedTime}
        nsfw={nsfw}
      />
      <Media.Item>
        <Content>
          <Heading size={3}>
            <NavLink to={`${DEVIATIONS_DETAILS.replace(ID_PARAM, id)}`}>
              {title}
            </NavLink>
          </Heading>
        </Content>
        <StatisticsLevel
          titleValuePairs={[['Views', views], ['Favourites', favourites], ['Comments', comments], ['Downloads', downloads]]}
        />
      </Media.Item>
      <Media.Item position="right">
        <Field>
          <Control>
            <Checkbox checked={selected} onChange={() => toggleSelectionHandler(id)} />
          </Control>
        </Field>
      </Media.Item>
    </Media>
  );
}

DeviationsPreview.propTypes = {
  ...DeviationThumbnail.PropTypes,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  toggleSelectionHandler: PropTypes.func.isRequired,
};
