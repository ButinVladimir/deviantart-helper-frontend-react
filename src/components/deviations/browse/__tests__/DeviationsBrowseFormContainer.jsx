import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import DeviationsBrowseFormContainer from '../DeviationsBrowseFormContainer';
import config from '../../../../config';
import { getConfigProvider } from '../../../shared/ConfigContext';
import createDefaultState from '../../../../redux/states/state';
import { FIELD_COMMENTS, ORDER_DESC } from '../../../../consts/sort';
import { SHOW_NSFW } from '../../../../consts/nsfw-options';
import {
  changeSortFieldActionCreator,
  changeSortOrderActionCreator,
  changeTitleActionCreator,
  changePublishedTimeBeginActionCreator,
  changePublishedTimeEndActionCreator,
  changeNsfwActionCreator,
  changeFilterByIdsActionCreator,
} from '../../../../redux/action-creators/deviations/browse/change-form-field-values';
import deviationsBrowseLoad, { deviationsBrowseLoadFirstPageActionCreator } from '../../../../redux/action-creators/deviations/browse/load-page';

jest.mock(
  '../../../../redux/action-creators/deviations/browse/change-form-field-values',
  () => ({
    __esModule: true,
    changeSortFieldActionCreator: jest.fn(() => ({ type: 'CHANGE_SORT_FIELD' })),
    changeSortOrderActionCreator: jest.fn(() => ({ type: 'CHANGE_SORT_ORDER' })),
    changeTitleActionCreator: jest.fn(() => ({ type: 'CHANGE_TITLE' })),
    changePublishedTimeBeginActionCreator: jest.fn(() => ({ type: 'CHANGE_PUBLISHED_TIME_BEGIN' })),
    changePublishedTimeEndActionCreator: jest.fn(() => ({ type: 'CHANGE_PUBLISHED_TIME_END' })),
    changeNsfwActionCreator: jest.fn(() => ({ type: 'CHANGE_NSFW' })),
    changeFilterByIdsActionCreator: jest.fn(() => ({ type: 'CHANGE_FILTER_BY_IDS' })),
  }),
);

jest.mock(
  '../../../../redux/action-creators/deviations/browse/load-page',
  () => ({
    __esModule: true,
    default: jest.fn(() => ({ type: 'LOAD_PAGE' })),
    deviationsBrowseLoadFirstPageActionCreator: jest.fn(() => ({ type: 'SUBMIT_FORM' })),
  }),
);

describe('DeviationsBrowseFormContainer', () => {
  const state = {
    ...createDefaultState(),
    deviations: {
      browse: {
        page: 1,
        pageCount: 3,
        sortField: FIELD_COMMENTS,
        sortOrder: ORDER_DESC,
        title: 'Title',
        publishedTimeBegin: 10001,
        publishedTimeEnd: 20001,
        nsfw: SHOW_NSFW,
        filterByIds: true,
        pageLoading: false,
        showPagination: true,
      },
    },
  };

  beforeEach(() => {
    changeSortFieldActionCreator.mockClear();
    changeSortOrderActionCreator.mockClear();
    changeTitleActionCreator.mockClear();
    changePublishedTimeBeginActionCreator.mockClear();
    changePublishedTimeEndActionCreator.mockClear();
    changeNsfwActionCreator.mockClear();
    changeFilterByIdsActionCreator.mockClear();
    deviationsBrowseLoad.mockClear();
    deviationsBrowseLoadFirstPageActionCreator.mockClear();
  });

  it('can be rendered correctly', () => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const ConfigProvider = getConfigProvider();
    const wrapper = mount(
      <ConfigProvider value={config}>
        <Provider store={store}>
          <DeviationsBrowseFormContainer />
        </Provider>
      </ConfigProvider>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('Form controls', () => {
    let wrapper = null;
    let store;

    beforeEach(() => {
      const mockStore = configureStore([thunk]);
      store = mockStore(state);

      const ConfigProvider = getConfigProvider();
      wrapper = mount(
        <ConfigProvider value={config}>
          <Provider store={store}>
            <DeviationsBrowseFormContainer />
          </Provider>
        </ConfigProvider>,
      );
    });

    it('can handle changing sort field', () => {
      wrapper.find('select#sort-field').simulate('change');

      expect(changeSortFieldActionCreator).toHaveBeenCalled();
      expect(store.getActions()).toEqual([{ type: 'CHANGE_SORT_FIELD' }]);
    });

    it('can handle changing sort order', () => {
      wrapper.find('select#sort-order').simulate('change');

      expect(changeSortOrderActionCreator).toHaveBeenCalled();
      expect(store.getActions()).toEqual([{ type: 'CHANGE_SORT_ORDER' }]);
    });

    it('can handle changing title', () => {
      wrapper.find('input#title').simulate('change');

      expect(changeTitleActionCreator).toHaveBeenCalled();
      expect(store.getActions()).toEqual([{ type: 'CHANGE_TITLE' }]);
    });

    it('can handle changing nsfw', () => {
      wrapper.find('select#nsfw').simulate('change');

      expect(changeNsfwActionCreator).toHaveBeenCalled();
      expect(store.getActions()).toEqual([{ type: 'CHANGE_NSFW' }]);
    });

    it('can handle changing filter by ids', () => {
      wrapper.find('input#filter-by-ids').simulate('change');

      expect(changeFilterByIdsActionCreator).toHaveBeenCalled();
      expect(store.getActions()).toEqual([{ type: 'CHANGE_FILTER_BY_IDS' }]);
    });

    it('can handle submitting', () => {
      wrapper.find('button#submit-button').simulate('click');

      expect(deviationsBrowseLoadFirstPageActionCreator).toHaveBeenCalledWith(config);
      expect(store.getActions()).toEqual([{ type: 'SUBMIT_FORM' }]);
    });

    it('can handle loading page', () => {
      wrapper.find('ul.pagination-list a.pagination-link').at(0).simulate('click');

      expect(deviationsBrowseLoad).toHaveBeenCalledWith(0, config);
      expect(store.getActions()).toEqual([{ type: 'LOAD_PAGE' }]);
    });
  });
});
