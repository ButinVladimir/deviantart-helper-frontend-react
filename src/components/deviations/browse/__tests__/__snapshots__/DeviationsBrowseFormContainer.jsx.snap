// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsBrowseFormContainer can be rendered correctly 1`] = `
<section
  className="section"
  style={Object {}}
>
  <div
    className="container"
    style={Object {}}
  >
    <div
      className="content"
      style={Object {}}
    >
      <h1
        className="title is-2"
      >
        Browse deviations
      </h1>
    </div>
    <div
      className="field"
      style={Object {}}
    >
      <label
        className="label"
        htmlFor={null}
        style={Object {}}
      >
        Title
      </label>
      <div
        className="control"
        style={Object {}}
      >
        <input
          className="input"
          disabled={false}
          id="title"
          name="title"
          onChange={[Function]}
          placeholder=""
          readOnly={false}
          style={Object {}}
          type="text"
          value="Title"
        />
      </div>
    </div>
    <div
      className="columns is-multiline"
      style={Object {}}
    >
      <div
        className="column"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            Published time range beginning
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <div>
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete="off"
                  className="input"
                  disabled={false}
                  id="published-time-begin"
                  name="publishedTimeBegin"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value="01.01.1970"
                />
                <button
                  className="react-datepicker__close-icon"
                  onClick={[Function]}
                  tabIndex={-1}
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="column"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            Published time range end
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <div>
              <div
                className="react-datepicker__input-container"
              >
                <input
                  autoComplete="off"
                  className="input"
                  disabled={false}
                  id="published-time-end"
                  name="publishedTimeEnd"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value="01.01.1970"
                />
                <button
                  className="react-datepicker__close-icon"
                  onClick={[Function]}
                  tabIndex={-1}
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="column"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            Sort by
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <div
              className="select"
              style={Object {}}
            >
              <select
                disabled={false}
                id="sort-field"
                multiple={false}
                name="sortField"
                onChange={[Function]}
                readOnly={false}
                value="comments"
              >
                <option
                  key="title"
                  value="title"
                >
                  Title
                </option>
                <option
                  key="publishedTime"
                  value="publishedTime"
                >
                  Published time
                </option>
                <option
                  key="views"
                  value="views"
                >
                  Views
                </option>
                <option
                  key="favourites"
                  value="favourites"
                >
                  Favourites
                </option>
                <option
                  key="comments"
                  value="comments"
                >
                  Comments
                </option>
                <option
                  key="downloads"
                  value="downloads"
                >
                  Downloads
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        className="column is-narrow"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            Sort order
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <div
              className="select"
              style={Object {}}
            >
              <select
                disabled={false}
                id="sort-order"
                multiple={false}
                name="sortOrder"
                onChange={[Function]}
                readOnly={false}
                value={-1}
              >
                <option
                  key="1"
                  value={1}
                >
                  Ascending
                </option>
                <option
                  key="-1"
                  value={-1}
                >
                  Descending
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="columns is-multiline"
      style={Object {}}
    >
      <div
        className="column"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            NSFW
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <div
              className="select"
              style={Object {}}
            >
              <select
                disabled={false}
                id="nsfw"
                multiple={false}
                name="nsfw"
                onChange={[Function]}
                readOnly={false}
                value="SHOW_NSFW"
              >
                <option
                  key="SHOW_ALL"
                  value="SHOW_ALL"
                >
                  Show everything
                </option>
                <option
                  key="SHOW_NON_NSFW"
                  value="SHOW_NON_NSFW"
                >
                  Show only non-NSFW
                </option>
                <option
                  key="SHOW_NSFW"
                  value="SHOW_NSFW"
                >
                  Show only NSFW
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        className="column is-narrow"
        style={Object {}}
      >
        <div
          className="field"
          style={Object {}}
        >
          <label
            className="label"
            htmlFor={null}
            style={Object {}}
          >
            Filter by selected
          </label>
          <div
            className="control"
            style={Object {}}
          >
            <label
              className="checkbox"
              disabled={false}
              style={Object {}}
            >
              <input
                checked={true}
                disabled={false}
                id="filter-by-ids"
                name="filterbyselected"
                onChange={[Function]}
                type="checkbox"
                value=""
              />
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="columns is-multiline"
      style={Object {}}
    >
      <div
        className="column"
        style={Object {}}
      >
        <nav
          aria-label="pagination"
          className="pagination"
          id="pagination"
        >
          <a
            className="pagination-previous"
            disabled={false}
            onClick={[Function]}
            role="button"
            tabIndex={0}
            title="This is the first page"
          >
            Previous
          </a>
          <a
            className="pagination-next"
            disabled={false}
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Next
          </a>
          <ul
            className="pagination-list"
          >
            <li
              key="1"
            >
              <a
                aria-current="page"
                aria-label="Page 1"
                className="pagination-link"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                1
              </a>
            </li>
            <li
              key="2"
            >
              <a
                aria-current="page"
                aria-label="Page 2"
                className="pagination-link is-current"
                role="button"
                tabIndex={0}
              >
                2
              </a>
            </li>
            <li
              key="3"
            >
              <a
                aria-current="page"
                aria-label="Page 3"
                className="pagination-link"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                3
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className="column is-narrow"
        style={Object {}}
      >
        <div
          className="field is-grouped is-grouped-right"
          style={Object {}}
        >
          <div
            className="control"
            style={Object {}}
          >
            <button
              className="is-primary button"
              disabled={false}
              id="submit-button"
              onClick={[Function]}
              style={Object {}}
              tabIndex={0}
            >
              Refresh
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
